src/core2/graph.js

```diff
@@ -369,17 +382,115 @@ export class Graph {
   }
 
   static fromJSON(plugins: Plugins, json: Compatible<GraphJSON>): Graph {
-    const _ = {plugins, json};
-    throw new Error("Graphv2 is not yet implemented");
+    const compatJson: GraphJSON = fromCompat(
+      {type: COMPAT_TYPE, version: COMPAT_VERSION},
+      json
+    );
+    const result = new Graph(plugins);
+    const pluginMap = createPluginMap(plugins);
+    compatJson.nodes.forEach((rawPartialNode) => {
+      if ("address" in rawPartialNode) {
+        const partialNode: {|+address: Address|} = (rawPartialNode: any);
+        result._addNodeAddress(partialNode.address);
+      } else {
+        const partialNode: {|
+          +payload: any,
+          +pluginName: string,
+        |} = (rawPartialNode: any);
+        const pluginName: string = partialNode.pluginName;
+        const payload = findHandler(pluginMap, pluginName).createPayload(
+          partialNode.payload
+        );
+        result.addNode(payload);
+      }
+    });
+    AddressMap.fromJSON(compatJson.edges)
+      .getAll()
+      .forEach((indexedEdge) => {
+        result._addIndexedEdge(indexedEdge);
+      });
+    return result;
   }
 
   toJSON(): Compatible<GraphJSON> {
-    throw new Error("Graphv2 is not yet implemented");
+    const partialNodes: {|
+      key: string,
+      oldIndex: Integer,
+      data:
+        | {|+address: Address|}
+        | {|+payload: NodePayload, +pluginName: string|},
+    |}[] = this._nodes
+      .map((maybeNode, oldIndex) => {
+        const key = stringify(maybeNode.address);
+
```

Comment:
nit: why this blankline?

Reply:
It does look a bit out of place, doesn’t it? I’ll remove it.